modules/widget-visibility/widget-conditions.php

```diff
@@ -39,7 +39,7 @@ public static function widget_conditions_options_echo( $major = '', $minor = ''
 				<option value=""><?php _e( 'All category pages', 'jetpack' ); ?></option>
 				<?php
 
-				$categories = get_categories( array( 'number' => 1000, 'orderby' => 'count', 'order' => 'DESC' ) );
+				$categories = get_categories( array( 'number' => 1000, 'orderby' => 'count', 'order' => 'DESC', 'hide_empty' => apply_filters( 'jetpack_widget_conditions_show_empty_terms' , false , $major , $minor ) ) );
```

Comment:
the filter name is confusing.  I'd expect `show_empty_terms` to want `true` in order to show.  `jetpack_widget_conditions_hide_empty_terms` is probably more fitting.